client/js/helpers/contextMenu.js

```diff
@@ -252,22 +252,28 @@ export function generateUserContextMenu($root, channel, network, user) {
 		return items;
 	}
 
-	// Names of the modes we are able to change
-	const modes = {
-		"~": ["owner", "q"],
-		"&": ["admin", "a"],
-		"@": ["operator", "o"],
-		"%": ["half-op", "h"],
-		"+": ["voice", "v"],
+	// Names of the standard modes we are able to change
+	const modeCharToName = {
+		"~": "owner",
+		"&": "admin",
+		"@": "operator",
+		"%": "half-op",
+		"+": "voice",
 	};
 
 	// Labels for the mode changes.  For example .rev(['admin', 'a']) => 'Revoke admin (-a)'
 	const modeTextTemplate = {
-		revoke: (m) => `Revoke ${m[0]} (-${m[1]})`,
-		give: (m) => `Give ${m[0]} (+${m[1]})`,
+		revoke(m) {
+			const name = modeCharToName[m.symbol] || "unknown mode";
+			return `Revoke ${name} (-${m.mode})`;
+		},
+		give(m) {
+			const name = modeCharToName[m.symbol] || "unknown mode";
+			return `Give ${name} (+${m.mode})`;
+		},
```

Comment:
I'd personally prefer if it just fell back to `Revoke -${m.mode}`.
Saying `Revoke/Give unknown mode` makes the context menu unnecessarily longer and doesn't looks as well.

Reply:
"give +$mode" is a redundant phrasing. The plus or minus already tells you what it is.

So either we should get rid of the prefix (give/revoke) or the symbols in my opinion.

I'm not hung up on the unknown mode thing, was just the first thing that came to mind

Reply:
The plus or minus bit with the actual mode character is basically letting the protocol leak through into the GUI. "Give" and "Revoke" (or "Remove") are the human-friendly expressions of those actions. It belongs in parentheses (current), or should be omitted if there mode's friendly name is known.

Strong agree on skipping the "unknown mode" name fallback, though. If the modechar is all that's available, having that fallback doesn't tell the user anything more than not having it.

Reply:
you still need some human parsable thing and that will still be constant.

How about "mode +Q" or what have you and correspondingly "mode -Q" ?

That (while not being symmetric to the human friendly names) at least makes it obvious what the options does.